### PR TITLE
feat: Add new field non_os_package_support to GHCR resource

### PR DESCRIPTION
**Pull request by dmurray-lacework**
***Issue***: https://lacework.atlassian.net/browse/ALLY-663

***Description:***
Add new field non_os_package_support for ghcr terraform resource

Signed-off-by: Darren Murray <darren.murray@lacework.net>

### CODE REVIEW

File: integration/resource_lacework_integration_ghcr_test.go [hunk 1]
```diff
@@ -29,12 +30,17 @@ func TestIntegrationGHCRCreate(t *testing.T) {
...
 
 		// Update Github Container Registry
 		terraformOptions.Vars["integration_name"] = "Github Container Registry Updated"
+		terraformOptions.Vars["non_os_packages"] = true
 
 		update := terraform.ApplyAndIdempotent(t, terraformOptions)
-		assert.Equal(t, "Github Container Registry Updated", GetIntegrationName(update))
+		updateData := GetContainerRegistryIntegration(update)
+		assert.Equal(t, "Github Container Registry Updated", updateData.Name)
+		assert.Equal(t, false, updateData.Data.NonOSPackageEval)
```
Comment (dmurray-lacework) [inline integration/resource_lacework_integration_ghcr_test.go:44]:
```suggestion
		assert.Equal(t, true, updateData.Data.NonOSPackageEval)
```

File: vendor/github.com/lacework/go-sdk/api/container_registries_ghcr.go [hunk 2]
```diff
@@ -60,6 +60,7 @@ type GhcrData struct {
 	LimitByLabel          []map[string]string `json:"limitByLabel"`
 	LimitByRep            []string            `json:"limitByRep"`
 	LimitNumImg           int                 `json:"limitNumImg"`
+	NonOSPackageEval 	  bool `json:"NON_OS_PACKAGE_EVAL,omitempty"`
```
Comment (dmurray-lacework) [inline vendor/github.com/lacework/go-sdk/api/container_registries_ghcr.go:63]:
```suggestion
	NonOSPackageEval 	  bool `json:"nonOsPackageEval,omitempty"`
```

File: examples/resource_lacework_integration_ghcr/variables.tf [hunk 3]
```diff
@@ -13,4 +13,7 @@ variable "password" {
 variable "ssl" {
   type = bool
 }
-
+variable "non_os_packages" {
```
Comment (afiune) [inline examples/resource_lacework_integration_ghcr/variables.tf:16]: Please update the rest and format all Terraform files
```suggestion
variable "non_os_package_support" {
```

**Review by afiune — CHANGES_REQUESTED** [review]

File: integration/resource_lacework_integration_ghcr_test.go [hunk 4]
```diff
@@ -18,8 +18,9 @@ func TestIntegrationGHCRCreate(t *testing.T) {
 		terraformOptions := terraform.WithDefaultRetryableErrors(t, &terraform.Options{
 			TerraformDir: "../examples/resource_lacework_integration_ghcr",
 			Vars: map[string]interface{}{
-				"username": creds.Username,
-				"ssl":      true,
+				"username":        creds.Username,
+				"ssl":             true,
+				"non_os_packages": false,
```
Comment (dmurray-lacework) [inline integration/resource_lacework_integration_ghcr_test.go:23]:
```suggestion
				"non_os_package_support": false,
```

File: integration/resource_lacework_integration_ghcr_test.go [hunk 5]
```diff
@@ -29,12 +30,17 @@ func TestIntegrationGHCRCreate(t *testing.T) {
...
 		// Create new Github Container Registry
 		create := terraform.InitAndApplyAndIdempotent(t, terraformOptions)
-		assert.Equal(t, "Github Container Registry Example", GetIntegrationName(create))
+		createData := GetContainerRegistryIntegration(create)
+		assert.Equal(t, "Github Container Registry Example", createData.Name)
+		assert.Equal(t, false, createData.Data.NonOSPackageEval)
 
 		// Update Github Container Registry
 		terraformOptions.Vars["integration_name"] = "Github Container Registry Updated"
+		terraformOptions.Vars["non_os_packages"] = true
```
Comment (dmurray-lacework) [inline integration/resource_lacework_integration_ghcr_test.go:39]:
```suggestion
		terraformOptions.Vars["non_os_package_support"] = true
```

**Review by afiune — APPROVED** [review]

**Pull request merged**